File: PozoleriaApp/app/src/main/java/com/example/pozoleria/CategoryAdapter.java
package com.example.pozoleria;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

public class CategoryAdapter extends RecyclerView.Adapter<CategoryAdapter.CategoryViewHolder> {

    private final List<CategoryItem> items;
    private final Context context;

    public CategoryAdapter(Context context, List<CategoryItem> items) {
        this.context = context;
        this.items = items;
    }

    @NonNull
    @Override
    public CategoryViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_category, parent, false);
        return new CategoryViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull CategoryViewHolder holder, int position) {
        final CategoryItem item = items.get(position);

        // TEXTO (usa getTitle porque así viene en tu CategoryItem)
        holder.txtCategoryName.setText(item.getTitle());

        // IMAGEN (usa getImageResId porque así viene en tu CategoryItem)
        holder.imgCategory.setImageResource(item.getImageResId());

        // Click por si quieres abrir productos
        holder.itemView.setOnClickListener(v -> {
            Intent intent = new Intent(context, ProductsActivity.class);
            intent.putExtra("categoryName", item.getTitle());
            context.startActivity(intent);
        });
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    static class CategoryViewHolder extends RecyclerView.ViewHolder {
        ImageView imgCategory;
        TextView txtCategoryName;

        public CategoryViewHolder(@NonNull View itemView) {
            super(itemView);

            // IDs EXACTOS según tu item_category.xml REAL
            imgCategory = itemView.findViewById(R.id.imgCategory);
            txtCategoryName = itemView.findViewById(R.id.txtCategoryName);
        }
    }
}

File: app/src/main/java/com/example/pozoleria/adapters/CategoryAdapter.kt
package com.example.pozoleria.adapters

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.pozoleria.ProductsActivity
import com.example.pozoleria.R
import com.example.pozoleria.models.CategoryItem

class CategoryAdapter(
    private val context: Context,
    private val categorias: List<CategoryItem>
) : RecyclerView.Adapter<CategoryAdapter.ViewHolder>() {

    inner class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val imgCategory: ImageView = view.findViewById(R.id.imgCategory)
        val txtCategoryName: TextView = view.findViewById(R.id.txtCategoryName)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val vista = LayoutInflater.from(context)
            .inflate(R.layout.item_category, parent, false)
        return ViewHolder(vista)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val item = categorias[position]

        holder.txtCategoryName.text = item.title
        holder.imgCategory.setImageResource(item.imageResId)

        holder.itemView.setOnClickListener {
            val intent = Intent(context, ProductsActivity::class.java)
            intent.putExtra(ProductsActivity.EXTRA_CATEGORY, item.title)
            context.startActivity(intent)
        }
    }

    override fun getItemCount(): Int = categorias.size
}
